lib/rules/selector-pseudo-class-allowed-list/index.mjs

```diff
@@ -26,55 +31,61 @@ const rule = (primary) => {
 			possible: [isString, isRegExp],
 		});
 
-		if (!validOptions) {
-			return;
-		}
-
-		root.walkRules((ruleNode) => {
-			if (!isStandardSyntaxRule(ruleNode)) {
-				return;
-			}
-
-			if (!ruleNode.selector.includes(':')) {
-				return;
-			}
-
-			parseSelector(getRuleSelector(ruleNode), result, ruleNode)?.walkPseudos((pseudoNode) => {
-				const value = pseudoNode.value;
-				const index = pseudoNode.sourceIndex;
-				const endIndex = index + pseudoNode.value.length;
+		if (!validOptions) return;
 
+		/**
+		 * @param {string} selector
+		 * @param {AtRule | Rule} node
+		 * @param {number} [offset]
+		 */
+		function check(selector, node, offset = 0) {
+			parseSelector(selector, result, node)?.walkPseudos(({ value, sourceIndex }) => {
 				// Ignore pseudo-elements
-				if (value.slice(0, 2) === '::') {
-					return;
-				}
+				if (value.slice(0, 2) === '::') return;
 
 				const name = value.slice(1);
 
-				if (matchesStringOrRegExp(name, primary)) {
-					return;
-				}
+				if (matchesStringOrRegExp(name, primary)) return;
+
+				if (matchesStringOrRegExp(vendor.unprefixed(name), primary)) return;
 
-				if (matchesStringOrRegExp(vendor.unprefixed(name), primary)) {
-					return;
-				}
+				const index = offset + sourceIndex;
 
 				report({
 					message: messages.rejected,
 					messageArgs: [value],
-					node: ruleNode,
+					node,
 					result,
 					ruleName,
 					index,
-					endIndex,
+					endIndex: index + value.length,
 				});
 			});
+		}
+
+		root.walkRules((ruleNode) => {
+			if (!isStandardSyntaxRule(ruleNode)) return;
+
+			const selector = getRuleSelector(ruleNode);
+
+			if (!selector.includes(':')) return;
+
+			check(selector, ruleNode);
+		});
+
+		root.walkAtRules('page', (atRuleNode) => {
+			if (!isStandardSyntaxAtRule(atRuleNode)) return;
```

Comment:
Can we add a test with a non-standard at rule?

Reply:
I think that codecov is being overzealous with these warnings.
i.e. it would be akin to adding cases to `isStandardSyntaxAtRule.test.mjs` directly

If you have a relevant test to add to `/selector-pseudo-class-allowed-list/__tests__/index.mjs` please provide it.

---

To be clear I think that new tests could probably be added to `isStandardSyntaxAtRule.test.mjs` but I won't add them in this PR.
e.g. [margin at-rules](https://drafts.csswg.org/css-page/#margin-at-rules)

Reply:
I think it could be fine to add a single test with non standard syntax.
Not to test the utility itself, but to encode that this rule doesn't support non-standard syntax in at rules.

But I don't think this is blocking :)